salt/modules/win_pkg.py

```diff
@@ -692,10 +789,15 @@ def genrepo(**kwargs):
                     ret,
                     successful_verbose
                     )
-    with salt.utils.fopen(repo_details.winrepo_file, 'w+b') as repo_cache:
-        repo_cache.write(msgpack.dumps(ret))
+    serial = salt.payload.Serial(__opts__)
+    # TODO: 2016.11 has PY2 mode as 'w+b' develop has 'w+' ? PY3 is 'wb+'
+    # also the reading of this is 'rb' in get_repo_data()
+    mode = 'w+' if six.PY2 else 'wb+'
+
+    with salt.utils.fopen(repo_details.winrepo_file, mode) as repo_cache:
+        repo_cache.write(serial.dumps(ret))
     # save reading it back again. ! this breaks due to utf8 issues
-    #__context__['winrepo.data'] = ret
+    # __context__['winrepo.data'] = ret
```

Comment:
Please just remove this if it's not being used.

Reply:
Updated the comment.